warehouse/warehouse.go

```diff
@@ -476,93 +412,25 @@ func (wh *HandleT) getPendingStagingFiles(warehouse warehouseutils.Warehouse) ([
 		panic(fmt.Errorf("query: %s failed with Error : %w", sqlStatement, err))
 	}
 
-	sqlStatement = fmt.Sprintf(`
-		SELECT
-		  id,
-		  location,
-		  status,
-		  first_event_at,
-		  last_event_at,
-		  metadata ->> 'source_batch_id',
-		  metadata ->> 'source_task_id',
-		  metadata ->> 'source_task_run_id',
-		  metadata ->> 'source_job_id',
-		  metadata ->> 'source_job_run_id',
-		  metadata ->> 'use_rudder_storage',
-		  metadata ->> 'time_window_year',
-		  metadata ->> 'time_window_month',
-		  metadata ->> 'time_window_day',
-		  metadata ->> 'time_window_hour',
-		  metadata ->> 'destination_revision_id'
-		FROM
-		  %[1]s ST
-		WHERE
-		  ST.id > %[2]v
-		  AND ST.source_id = '%[3]s'
-		  AND ST.destination_id = '%[4]s'
-		ORDER BY
-		  id ASC;
-`,
-		warehouseutils.WarehouseStagingFilesTable,
-		lastStagingFileID,
+	stagingFilesList, err := wh.stagingRepo.GetAfterID(
```

Comment:
Should we consider returning `[]*model.StagingFile` as we are doing pointer conversion later?

Reply:
@lvrach Any benefits of this?